arangod/Graph/Cache/RefactoredTraverserCache.cpp

```diff
@@ -120,75 +120,66 @@ bool RefactoredTraverserCache::appendEdge(EdgeDocumentToken const& idToken,
     TRI_ASSERT(col != nullptr);  // for maintainer mode
     return false;
   }
-
+  auto cb = IndexIterator::makeDocumentCallbackF(
+      [&](LocalDocumentId, VPackSlice edge) -> bool {
+        if (readType == EdgeReadType::ONLYID) {
+          if constexpr (std::is_same_v<ResultType, std::string>) {
+            // If we want to expose the ID, we need to translate the
+            // custom type Unfortunately we cannot do this in slice only
+            // manner, as there is no complete slice with the _id.
+            result = transaction::helpers::extractIdString(
+                _trx->resolver(), edge, VPackSlice::noneSlice());
+            return true;
+          }
+          edge = edge.get(StaticStrings::IdString).translate();
+        } else if (readType == EdgeReadType::ID_DOCUMENT) {
+          if constexpr (std::is_same_v<ResultType, velocypack::Builder>) {
+            TRI_ASSERT(result.isOpenObject());
+            TRI_ASSERT(edge.isObject());
+            // Extract and Translate the _key value
+            result.add(VPackValue(transaction::helpers::extractIdString(
+                _trx->resolver(), edge, VPackSlice::noneSlice())));
+            if (!_edgeProjections.empty()) {
+              VPackObjectBuilder guard(&result);
+              _edgeProjections.toVelocyPackFromDocument(result, edge, _trx);
+            } else {
+              result.add(edge);
+            }
+            return true;
+          } else {
+            // We can only inject key_value pairs into velocypack
+            TRI_ASSERT(false);
+          }
+        }
+        // NOTE: Do not count this as Primary Index Scan, we
+        // counted it in the edge Index before copying...
+        if constexpr (std::is_same_v<ResultType, aql::AqlValue>) {
+          if (!_edgeProjections.empty()) {
+            // TODO: This does one unnecessary copy.
+            // We should be able to move the Projection into the
+            // AQL value.
+            transaction::BuilderLeaser builder(_trx);
+            {
+              VPackObjectBuilder guard(builder.get());
+              _edgeProjections.toVelocyPackFromDocument(*builder, edge, _trx);
+            }
+            result = aql::AqlValue(builder->slice());
+          } else {
+            result = aql::AqlValue(edge);
+          }
+          result = aql::AqlValue(edge);
```

Comment:
I see that this is identical to the original code, but it seems that we are unconditionally clobbering `result` in line 170, even if it was set to something else in lines 166 or 168. This looks inefficient and wrong. Especially it looks like that the projections currently cause _extra_ work, and the full edge is copied out later anyway.

Reply:
I just comment out code above, if you know how to fix it, feel free to suggest

Reply:
We can leave it as you did it now in this PR, and fix it later when this PR is merged.